packages/database/src/modelClasses/MapOverlayGroupRelation.js

```diff
@@ -57,4 +58,84 @@ export class MapOverlayGroupRelationModel extends DatabaseModel {
       'map_overlay_group_id',
     );
   }
+
+  /**
+   * Overrides `DatabaseModel.find` to add `child_code` to the model. Note that this is a departure
+   * from the underlying database schema.
+   *
+   * @see generateInstance
+   */
+  async find(criteria, dbOptions = {}) {
+    const options = dbOptions;
+
+    // Left join with map_overlay and map_overlay_group
+    options.multiJoin = [
+      {
+        joinType: JOIN_TYPES.LEFT,
+        joinWith: RECORDS.MAP_OVERLAY,
+        joinCondition: [
+          `${RECORDS.MAP_OVERLAY_GROUP_RELATION}.child_id`,
+          `${RECORDS.MAP_OVERLAY}.id`,
+        ],
+      },
+      {
+        joinType: JOIN_TYPES.LEFT,
+        joinWith: RECORDS.MAP_OVERLAY_GROUP,
+        joinCondition: [
+          `${RECORDS.MAP_OVERLAY_GROUP_RELATION}.child_id`,
+          `${RECORDS.MAP_OVERLAY_GROUP}.id`,
+        ],
+      },
+    ];
+
+    // Add child code for both child type options
+    options.columns = [];
+    for (const { joinWith } of options.multiJoin) {
+      const column = `${joinWith}.code as ${
+        joinWith === RECORDS.MAP_OVERLAY ? 'childMapOverlayCode' : 'childMapOverlayGroupCode'
+      }`;
+      options.columns.push(column);
+    }
```

Comment:
A little bit conflicted about using a loop here, when it could just as easily be

```tsx
options.columns = [
  `${MAP_OVERLAY}.code as childMapOverlayCode`,
  `${MAP_OVERLAY_GROUP} as childMapOverlayGroupCode`,
]
```

Reply:
Refactored out